op-chain-ops/script/deploy.go

```diff
@@ -0,0 +1,330 @@
+package script
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/ethereum-optimism/optimism/op-chain-ops/foundry"
+	"github.com/ethereum-optimism/optimism/op-chain-ops/script/addresses"
+	"github.com/holiman/uint256"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+// ForgeScriptBackend is a minimal interface suitable for deploying & interacting with
+// foundry scripts on chain
+type ForgeScriptBackend interface {
```

Comment:
I assume this maps to `script.Host`?

Reply:
Yess, it's just wrapped into an interface to decouple the logic from the implementation